contracts/src/IVerifier.sol

```diff
@@ -0,0 +1,69 @@
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity 0.8.17;
+
+/**
+ * @title Interface of Verifier contract
+ * @author Seiya Kobayashi
+ */
+interface IVerifier {
+    /// @dev Holds hash value of node
```

Comment:
What about separate merkle tree implementation as library or another contract? or you may use this implementation. https://docs.openzeppelin.com/contracts/4.x/api/utils#MerkleProof